sc-poc/modules/pubsub/asyncapi.go

```diff
@@ -0,0 +1,243 @@
+package pubsub
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/spacecloud-io/space-cloud/managers/apis"
+)
+
+func (a *AsyncAPI) AddServer(name string, srv ServerItem) {
+	if a.Servers == nil {
+		a.Servers = make(map[string]*ServerItem)
+	}
+
+	a.Servers[name] = &srv
+}
+
+func (a *AsyncAPI) AddChannel(name string, channel ChannelItem) {
+	if a.Channels == nil {
+		a.Channels = make(map[string]*ChannelItem)
+	}
+
+	a.Channels[name] = &channel
+}
+
+func (a *App) generateASyncAPIDoc() *apis.API {
+	return &apis.API{
+		Name: "asyncapi",
+		Path: "/v1/api/asyncapi.json",
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Add general Info
+			asyncapi := AsyncAPI{}
```

Comment:
Compute this object in provision. Simply marshal and return it in this handler. I don't want you to compute the entire doc on every request